subsys/debug/etb_trace/etb_trace.c

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <debug/etb_trace.h>
+
+#include "etb_trace_private.h"
+
+/* Declaring as static variables to make them appear with the correct value in ELF file
+ * for simplified trace processing.
+ */
+
+/* Bit 3: Branch broadcast mode enabled. */
+static volatile uint32_t etm_trcconfigr = BIT(3);
+/* Set trace ID to 0x10  */
+static volatile uint32_t etm_trctraceidr = 0x10;
+
+static void etm_init(void)
+{
+	/* Disable ETM to allow configuration */
+	SET_REG(ETM_TRCPRGCTLR, 0);
+
+	/* Wait until ETM is idle and PM is stable */
+	while ((GET_REG(ETM_TRCSTATR) & (BIT(1) | BIT(0))) != (BIT(1) | BIT(0))) {
+	}
+
+	/* Configure the ETM */
+	SET_REG(ETM_TRCCONFIGR, etm_trcconfigr);
+
+	/* The trace unit can not stall the processor for instruction traces, at the risk of
+	 * losing traces.
+	 */
+	SET_REG(ETM_TRCSTALLCTLR, 0);
+
+	/* Global Timestamp Control Register to zero, no stamps included in the trace.  */
+	SET_REG(ETM_TRCTSCTLR, 0);
+
+	/* Set the trace stream ID */
+	SET_REG(ETM_TRCTRACEIDR, etm_trctraceidr);
+
+	/* Bit 0: Enable event 0.
+	 * Bit 9: Indicates the current status of the start/stop logic.
+	 * Bit 10: Always trace reset exceptions.
+	 * Bit 11: Always trace system error exceptions.
+	 */
+	SET_REG(ETM_TRCVICTLR, BIT(11) | BIT(10) | BIT(9) | BIT(0));
+
+	/* No events are configured */
```

Comment:
Some more explanation would be nice. What are these events? and why should they not be configured?

Reply:
According to the Technical Reference Manual, we have to write these registers as part of initialization. They configure features we don't use, so we just write 0 to disable them. Is it really necessary to go into details about these unused features?